src/main/python/calibration/parse_google_result.py

```diff
@@ -0,0 +1,215 @@
+    # typically 35 min - 1 h
+    # typically 35 min - 1 h 5 min
+    # typically 40 min - 1 h
+    # typically 40 min - 1 h 5 min
+    regex = r"(?:typically\s){0,1}([\d]{1,2}\sh\s){0,1}([\d]{1,2}\smin\s){0,1}-\s([\d]{1,2}\sh[\s]{0,1})([\d]{1,2}\smin){0,1}$"
+    matches = re.finditer(regex, travel_time_str)
+    enum = list(enumerate(matches, start=1))
+    if len(enum) == 1:
+        n, match = enum[0]
+        # group 1 - is the hour for lower bound
+        min_h = ifNoneThenEmptyString(match.group(1)).replace('h', '')
+        min_m = ifNoneThenEmptyString(match.group(2)).replace('min', '')
+        max_h = ifNoneThenEmptyString(match.group(3)).replace('h', '')
+        max_m = ifNoneThenEmptyString(match.group(4)).replace('min', '')
+        min_time = getTotalTimeInMinutes(min_m, min_h) * 60
+        max_time = getTotalTimeInMinutes(max_m, max_h) * 60
+        return (min_time, max_time)
+    else:
+        return None
+def third_attempt(travel_time_str):
+     # for a case like `typically 8min`
+    regex = r"(?:typically\s){0,1}([\d]{1,2})\smin$"
+    matches = re.finditer(regex, travel_time_str)
+    enum = list(enumerate(matches, start=1))
+    if len(enum) == 1:
+        n, match = enum[0]
+        min_time = int(match.group(1)) * 60
+        max_time = int(match.group(1)) * 60
+        return (min_time, max_time)
+    else:
+        return None
+def fourth_attempt(travel_time_str):
+     # for a case like `typically 1 h`
+    regex = r"(?:typically\s){0,1}([\d]{1,2})\sh$"
+    matches = re.finditer(regex, travel_time_str)
+    enum = list(enumerate(matches, start=1))
+    if len(enum) == 1:
+        n, match = enum[0]
+        min_time = int(match.group(1)) * 60 * 60
+        max_time = int(match.group(1)) * 60 * 60
+        return (min_time, max_time)
+    else:
+        return None
+def parse_travel_time(travel_time_str):
+    if isinstance(travel_time_str,float) and math.isnan(travel_time_str):
+        return (float('nan'), float('nan'))
+    temp = first_attempt(travel_time_str)
+    if temp is not None:
+        return temp
+    temp = second_attempt(travel_time_str)
+    if temp is not None:
+        return temp
+    temp = third_attempt(travel_time_str)
+    if temp is not None:
+        return temp
+    temp = fourth_attempt(travel_time_str)
+    if temp is not None:
+        return temp
+    else:
+        raise Exception("Cannot parse '%s' as travel time" % ((travel_time_str)))
+def parse_travel_distance(travel_distance):
+    if isinstance(travel_distance,float) and math.isnan(travel_distance):
+        return float('nan')
+    elif "km" in travel_distance:
+        return float(travel_distance.replace('km', '')) * 1000
+    elif "miles" in travel_distance:
+        # Miles to meteres
+        return float(travel_distance.replace('miles', '')) * 1.60934 * 1000
+    elif "mile" in travel_distance:
+        # Miles to meteres
+        return float(travel_distance.replace('mile', '')) * 1.60934 * 1000
```

Comment:
this method can be simplified

Reply:
Sure it can, but I wrote this code during debugging the output of Google, so at least for me it makes sense to leave it as is